src/dataset/tag.rs

```diff
@@ -0,0 +1,135 @@
+use std::fmt;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
```

Comment:
Do we need the "Debug" here?

Reply:
It is a bit of a courtesy to have a `Debug` impl for "vocabulary" types. It is very likely that one will want to do something like `dbg!(dataset);` during development in the future and recursively chasing down any types missing `#[derive(Debug)]` is rather cumbersome. (It also does not cost us anything at runtime if the methods are never called, just a bit of compile time to generate them.)